trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/CHandler.java

```diff
@@ -835,7 +836,8 @@ public Result visit(final IDispatcher main, final IASTBinaryExpression node) {
 			final ExpressionResult rr = mExprResultTransformer.transformSwitchRexBoolToInt(rightOperand, loc, node);
 			final ExpressionResult result =
 					mCExpressionTranslator.handleMultiplicativeOperation(loc, node.getOperator(), rl, rr);
-			return makeAssignment(loc, leftOperand.getLrValue(), Collections.emptyList(), result, node);
+			return makeAtomicAssignmentIfNecessary(loc, leftOperand.getLrValue(),
+					DataStructureUtils.concat(leftOperand.getStatements(), rr.getStatements()), result, node);

```

Comment:
Why do you use `leftOperand.getStatements()` rather than `rl.getStatements()` here?

Reply:
Because the switch to RValue (i.e., read of the heap variable if on heap) should be executed atomically.

Reply:
But I will add a comment for this.